src/main/java/org/junitpioneer/internal/PioneerUtils.java

```diff
@@ -82,6 +84,22 @@ private PioneerUtils() {
 		return method;
 	}
 
+	/**
+	 * Find all (parent) {@code ExtensionContext}s via {@link ExtensionContext#getParent()}.
+	 *
+	 * @param context the context for which to find all (parent) contexts; never {@code null}
+	 * @return a list of all contexts, including the given context; never {@code null} or empty
+	 */
+	public static List<ExtensionContext> findAllContexts(ExtensionContext context) {
+		List<ExtensionContext> contexts = new ArrayList<>();
+		Optional<ExtensionContext> currentContext = Optional.of(context);
+		do {
+			currentContext.ifPresent(contexts::add);
```

Comment:
But `currentContext` is always `present`...

Reply:
Indeed, let's use recursion then. 😬